From 2e4e5d3afdfca930082e420e45153493696cd69a Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Tue, 31 Jul 2018 21:01:43 +0200
Subject: [PATCH 6/9] sdk: Add minimal LineageSettingsService

* This service is currently used for loading custom hostname,
  it may be used for more than that in the future.

Change-Id: I9b4da124b54bdc28544becb2ad25b7b43d46f123
---
 .../internal/LineageSettingsService.java      | 55 +++++++++++++++++++
 lineage/res/res/values/config.xml             |  1 +
 .../app/LineageContextConstants.java          |  8 +++
 3 files changed, 64 insertions(+)
 create mode 100644 lineage/lib/main/java/org/lineageos/platform/internal/LineageSettingsService.java

diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/LineageSettingsService.java b/lineage/lib/main/java/org/lineageos/platform/internal/LineageSettingsService.java
new file mode 100644
index 0000000..2655110
--- /dev/null
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/LineageSettingsService.java
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2018 The LineageOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.lineageos.platform.internal;
+
+import android.content.Context;
+import android.os.SystemProperties;
+
+import lineageos.app.LineageContextConstants;
+import lineageos.providers.LineageSettings;
+
+/** @hide */
+public class LineageSettingsService extends LineageSystemService {
+
+    private static final String TAG = LineageSettingsService.class.getSimpleName();
+
+    private final Context mContext;
+
+    public LineageSettingsService(Context context) {
+        super(context);
+        mContext = context;
+    }
+
+    @Override
+    public String getFeatureDeclaration() {
+        return LineageContextConstants.Features.SETTINGS;
+    }
+
+    @Override
+    public void onBootPhase(int phase) {
+        if (phase == PHASE_BOOT_COMPLETED) {
+            // Load custom hostname
+            String hostname = LineageSettings.Secure.getString(mContext.getContentResolver(),
+                    LineageSettings.Secure.DEVICE_HOSTNAME);
+            SystemProperties.set("net.hostname", hostname);
+        }
+    }
+
+    @Override
+    public void onStart() {
+    }
+}
diff --git a/lineage/res/res/values/config.xml b/lineage/res/res/values/config.xml
index 4a57ce1..e049722 100644
--- a/lineage/res/res/values/config.xml
+++ b/lineage/res/res/values/config.xml
@@ -110,6 +110,7 @@
         <item>org.lineageos.platform.internal.LineageAudioService</item>
         <item>org.lineageos.platform.internal.StyleInterfaceService</item>
         <item>org.lineageos.platform.internal.TrustInterfaceService</item>
+        <item>org.lineageos.platform.internal.LineageSettingsService</item>
     </string-array>
 
     <!-- The LineageSystemServer class that is invoked from Android's SystemServer -->
diff --git a/sdk/src/java/lineageos/app/LineageContextConstants.java b/sdk/src/java/lineageos/app/LineageContextConstants.java
index 33fd85b..8a210cd 100644
--- a/sdk/src/java/lineageos/app/LineageContextConstants.java
+++ b/sdk/src/java/lineageos/app/LineageContextConstants.java
@@ -185,5 +185,13 @@ public final class LineageContextConstants {
          */
         @SdkConstant(SdkConstant.SdkConstantType.FEATURE)
         public static final String TRUST = "org.lineageos.trust";
+
+        /**
+         * Feature for {@link PackageManager#getSystemAvailableFeatures} and
+         * {@link PackageManager#hasSystemFeature}: The device includes the lineage settings service
+         * utilized by the lineage sdk.
+         */
+        @SdkConstant(SdkConstant.SdkConstantType.FEATURE)
+        public static final String SETTINGS = "org.lineageos.settings";
     }
 }
-- 
2.17.1

